containers/release-candidate/Dockerfile

```diff
@@ -15,20 +15,61 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM registry.fedoraproject.org/fedora-minimal:36
+FROM registry.fedoraproject.org/fedora:36
 
 WORKDIR /root
 
-COPY setup-container.sh /usr/bin/
-RUN /usr/bin/setup-container.sh
+# Delete cruft fedora leaves in the homedir
+RUN rm /root/*
 
-# Needed to get WiX to run in wine on Linux. See wix_wine.sh for more details
-# on why we need to do this and how it works
+# Install dependencies, explicitly not enabling the updates repo so we are
+# pinned at a particular fedora release.
+RUN \
+  dnf -y --quiet --repo=fedora install \
+    clang \
+    git \
+    java-1.8.0-devel \
+    llvm \
+    mxml-devel \
+    npm \
+    pinentry \
+    rpm-build \
+    rpm-sign \
+    subversion \
+    unzip \
+    vim-minimal \
+    wine \
+    winetricks && \
+  dnf clean all
+
+# Enable SBT pgp plugin
+COPY src/plugins.sbt /root/.sbt/1.0/plugins/
+
+# Install wix, including changes to allow WiX to run in wine on Linux. See
+# src/wix_wine.sh for more details on why we need to do this and how it works
+RUN \
+  curl -sS -L https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311-binaries.zip -o wix311-binaries.zip && \
+  echo "6fd961c85e1e6adafb99ef50c9659e3eb83c84ecaf49f523e556788845b206e1857aba2c39409405d4cda1df9b30a552ce5aab808be5f8368a37a447d78d1a05 wix311-binaries.zip" | sha512sum --quiet -c - && \
+  mkdir /opt/wix311 && \
+  unzip -q wix311-binaries.zip -d /opt/wix311/ && \
+  rm wix311-binaries.zip
 RUN mv /opt/wix311/{candle.exe,real-candle.exe}
 RUN mv /opt/wix311/{light.exe,real-light.exe}
-COPY wix_wine.sh /opt/wix311/candle.exe
-COPY wix_wine.sh /opt/wix311/light.exe
+COPY src/wix_wine.sh /opt/wix311/candle.exe
+COPY src/wix_wine.sh /opt/wix311/light.exe
+
+# Install a pinned version of sbt, needed because the spt-rpm repository
+# frequently adds new versions.
+RUN \
+  dnf -y --quiet --repofrompath=sbt-rpm,https://repo.scala-sbt.org/scalasbt/rpm --repo=sbt-rpm --nogpgcheck install \
+    sbt-1.7.1 && \
```

Comment:
When and how do you propose we bump these pinned versions of sbt and yarn?  Add a bullet item to the release checklist to bump sbt and yarn to their latest versions whenever we make a new release?

Reply:
Good point. Dependabot knows how to update Dockerfile base images, but I don't think it is smart enough to update the npm/dnf commands. This might be something we just need to check during releases, or try to remember to do when we get related pull requests.

Note that it probably won't matter too much in practice if we don't update these. All dependency versions are defined in the repos (including SBT, not sure about yarn).

Reply:
I've added a line in the "prepare" section of the release candidate workflow to check and update the versions. I don't think there's an good automated way of handling this.